src/parsers/STATICPOOLparser.py

```diff
@@ -0,0 +1,149 @@
+
+        self.input_col_idxs = [
+            self.mapping_column_indices[col_idx] for col_idx in self.input_keys
+        ]
+        self.output_col_idxs = [
+            self.mapping_column_indices[col_idx] for col_idx in self.target_keys
+        ]
+
+        self.train, self.valid, self.test, self.pool = data_split(
+            self.data, **self.data_args
+        )
+        self.train = self.train.sample(0)
+        self.train = torch.tensor(self.train.values)
+        self.valid = torch.tensor(self.valid.values)
+        self.test = torch.tensor(self.test.values)
+        self.pool = torch.tensor(self.pool.values)
+        # self.train, self.valid, self.test, self.pool, self.scaler = apply_scaler(self.train, self.valid, self.test, self.pool, scaler=None, op='transform')
+
+    def gather_data_from_storage(self, data_path) -> pd.DataFrame:
+        extension = data_path.split(".")[-1]
+        if extension == "csv":
+            data = CSVLoader(data_path=data_path).load_data()
+        elif extension == "pkl":
+            data = PickleLoader(data_path=data_path).load_data()
+        elif extension == "h5":
+            data = HDFLoader(data_path=data_path).load_data()
+        else:
+            raise ValueError(
+                f"Extension {extension} not allowed. Please use any of {['csv','h5','pkl']}"
+            )
+        return data
+
+    def get_column_idx_mapping(self) -> dict:
+        return {
+            col: idx
+            for col, idx in zip(self.data.columns, range(len(self.data.columns)))
+        }
+
+    def get_train_valid(self) -> tuple[torch.Tensor, torch.Tensor]:
+        # TODO this should be a method in the base Parser class
+        return self.train, self.valid
+
+    def scale_train_val_test_pool(
+        self,
+        train: torch.Tensor,
+        valid: torch.Tensor,
+        test: torch.Tensor,
+        pool: torch.Tensor,
+    ) -> tuple[torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor]:
+        train, valid, test, pool, scaler = apply_scaler(
+            train, valid, test, pool, scaler=None, op="transform"
+        )
+        return train, valid, test, pool
+
+    def get_train_valid_test_pool(
+        self,
+    ) -> tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
+        return self.train, self.valid, self.test, self.pool
+
+    def get_train_valid_datasplit_from_self(
+        self,
+    ) -> tuple[tuple[torch.Tensor, torch.Tensor], tuple[torch.Tensor, torch.Tensor]]:
+        """returns normalized (x_train, y_train), (x_valid, y_valid)"""
+        x_train, y_train = (
+            self.train[:, self.input_col_idxs].float(),
+            self.train[:, self.output_col_idxs].float(),
+        )
+        x_valid, y_valid = (
+            self.valid[:, self.input_col_idxs].float(),
+            self.valid[:, self.output_col_idxs].float(),
+        )
+
+        return (x_train, y_train), (x_valid, y_valid)
+
+    def make_train_valid_datasplit(
```

Comment:
specify in the docstring that the `train` and `valid` tensors are normalised already

Reply:
I don't think that is necessary as the function takes arbitrary train/valid tensors, so one could run `train, valid = self.get_train_valid() -> self.make_train_valid_dataset(train, valid)`. But we should consider documenting the executor workflow a bit more.